src/Functions/FunctionDateOrDateTimeToDateTimeOrDateTime64.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+#include <Functions/IFunctionDateOrDateTime.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+template <typename Transform>
+class FunctionDateOrDateTimeToDateTimeOrDateTime64 : public IFunctionDateOrDateTime<Transform>, WithContext
+{
+public:
+    const bool enable_extended_results_for_datetime_functions = false;
```

Comment:
Minor: `enable_extended_results_for_datetime_functions` could be in the private section (same applies to "FunctionDateOrDateTimeToDateOrDate32.h")

Reply:
Done